### DISCUSSION TITLE
Adding Cookiebot script to <head/> with <ClientOnly> does not execute the script

### DISCUSSION THREAD

**Discussion by khanharis87** [remix-run/remix#7499] [Q&A]
I am trying to make Cookiebot work , which is done by adding the script in <head> and that loads the cookie banner. Adding just the <script/> causes a lot of errors, and most of them related to Hydration, also the banner sometime is visible and sometimes it is not.

Looking at internet for answers I came across  <ClientOnly>  solution provided by remix-utils library, this does the the job of adding the script tag only on client but the script does not execute. I am  a bit lost on why it might be happening and looking for guidance on where to start looking for solution.

**Comment by sergiodxa** [comment]
React render scripts tags using `innerHTML` this prevents the execution of scripts, instead create a script element with `document.createElement("script")` inside an effect and later attach it to the DOM, this will achieve the same behavior you want of using the script only after hydration.
